### PR TITLE
virt-test : Add windows 10 support

### PR DESCRIPTION
**Pull request by vipmike007**
Add configuration and xml file for windows 10

Signed-off-by: Mike Cao <bcao@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
@vipmike007 
do we need special autoinstall.xml for ovmf test ?

File: shared/cfg/guest-os/Windows/Win10/i386.cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+- i386:
+    vm_arch_name = i686
+    image_name += -32
+    unattended_install.cdrom, whql.support_vm_install, svirt_install:
+       cdrom_cd1 = isos/ISO/Win10/en_windows_10_enterprise_x86_dvd_6851156.iso
+        md5sum_cd1 = 68162933e551e55779504dacdfb0c5ae
+        md5sum_1m_cd1 = b44af2fdb8c39bc972f416bd3616566
```
Comment (CongSmile) [inline shared/cfg/guest-os/Windows/Win10/i386.cfg]: it's 31 bit, miss one bit here.

**Comment by CongSmile** [comment]
Not all versions support win10, could you also check it?

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

File: shared/cfg/guest-os/Windows/Win10.cfg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+- Win10:
+    os_variant = win10
+    image_name = images/win10
+    shutdown_command = shutdown /s /f /t 0
```
Comment (xutian) [inline shared/cfg/guest-os/Windows/Win10.cfg]: @vipmike007 
shutdown/reboot command should be in windows.cfg?

File: shared/cfg/guest-os/Windows/Win10/i386.cfg [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    vm_arch_name = i686
+    image_name += -32
+    unattended_install.cdrom, whql.support_vm_install, svirt_install:
+       cdrom_cd1 = isos/ISO/Win10/en_windows_10_enterprise_x86_dvd_6851156.iso
+        md5sum_cd1 = 68162933e551e55779504dacdfb0c5ae
+        md5sum_1m_cd1 = b44af2fdb8c39bc972f416bd3616566f
+        unattended_file = unattended/win10-32-autounattend.xml
+        floppies = "fl"
+        floppy_name = images/win10-32/answer.vfd
+        extra_cdrom_ks:
```
Comment (xutian) [inline shared/cfg/guest-os/Windows/Win10/i386.cfg]: this part of configuration should be in Win10.cfg

File: shared/cfg/guest-os/Windows/Win10/x86_64.cfg [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        md5sum_1m_cd1 = 0e246b56b7b63f0526c3367eb160ff7d
+        passwd = 1q2w3eP
+    unattended_install.cdrom, whql.support_vm_install, svirt_install:
+        cdrom_cd1 = isos/windows/en_windows_10_enterprise_x64_dvd_6851151.iso
+        md5sum_cd1 = a67722adfaf209c72eacb3ab910ee65e
+        md5sum_1m_cd1 = 0e246b56b7b63f0526c3367eb160ff7d
+        unattended_file = unattended/win10-64-autounattend.xml
+        floppies = "fl"
+        floppy_name = images/win10-64/answer.vfd
+        extra_cdrom_ks:
```
Comment (xutian) [inline shared/cfg/guest-os/Windows/Win10/x86_64.cfg]: after comment@11line, please remove this part of configuration.

File: shared/cfg/guest-os/Windows/Win10/x86_64.cfg [hunk 5]
```diff
@@ -0,0 +1,26 @@
+- x86_64:
+    image_name += -64
+    vm_arch_name = x86_64
+    install:
+        cdrom_cd1 = isos/ISO/Win10/en_windows_10_enterprise_x64_dvd_6851151.iso
+        md5sum_cd1 = a67722adfaf209c72eacb3ab910ee65e
+        md5sum_1m_cd1 = 0e246b56b7b63f0526c3367eb160ff7d
+        passwd = 1q2w3eP
+    unattended_install.cdrom, whql.support_vm_install, svirt_install:
+        cdrom_cd1 = isos/windows/en_windows_10_enterprise_x64_dvd_6851151.iso
```
Comment (xutian) [inline shared/cfg/guest-os/Windows/Win10/x86_64.cfg]: why ISO path different with line#5?

**Comment by xutian** [comment]
@vipmike007 

if windows10's  shutdown/reboot command not different with other windows release, please move it to windows.cfg. and  remove these command in special cfg too.

Thanks,
Xu

File: shared/cfg/guest-os/Windows/Win10.cfg (hunk 2 above)
Comment (vipmike007) [inline shared/cfg/guest-os/Windows/Win10.cfg]: Hi, I referring to the win8/win2012 setup.
I will remove them all

File: shared/cfg/guest-os/Windows/Win10/x86_64.cfg (hunk 5 above)
Comment (vipmike007) [inline shared/cfg/guest-os/Windows/Win10/x86_64.cfg]: This line is copy from windows 8 ,and I have another patch in internal for redhat only path.
Mike

**Comment by vipmike007** [comment]
Hi, pls recheck.

File: shared/cfg/guest-os/Windows/Win10/i386.cfg [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            virtio_network_path = 'A:\i386\Win8'
+            virtio_scsi_path = 'A:\i386\Win8'
+            extra_cdrom_ks:
+                virtio_storage_path = 'G:\viostor\w8\x86'
+                virtio_network_path = 'G:\NetKVM\w8\x86'
+                virtio_scsi_path = 'G:\vioscsi\w8\x86'
+                virtio_qxl_installer_path = E:\\install_driver.bat G:\\qxl\\Win8\\x86
+                virtio_balloon_installer_path = E:\\install_driver.bat G:\\Balloon\\w8\\x86
+                cd_format = ide
+            q35:
```
Comment (suqinhuang) [inline shared/cfg/guest-os/Windows/Win10/i386.cfg]: wrong alignment

**Comment by suqinhuang** [comment]
pls add xml file to ovmf as well

**Comment by lmr** [comment]
Hi @vipmike007. As you might know, we're moving virt-test to avocado-vt, so I please ask you to consider re-implementing this functionality in the later. The new repo is:

https://github.com/avocado-framework/avocado-vt

Please keep in mind the following guidelines when creating the new PR:

1) Most of the time you can pick the patches of your original PR and merge them on a new branch with minimal changes.
3) Don't use autotest APIs, since we're moving away from using them. Please review the code and change things like `utils.system` -> `process.system`, `utils.run` -> `process.run`, so on and so forth.

Thanks, and I'm sorry for any inconveniences we might have caused your team.

**Pull request closed without merging**